app/Renderers/Layouts/Mention/Mention.php

```diff
@@ -17,13 +17,24 @@ public function columns(): array
             'name' => [
                 'key' => 'name',
                 'label' => 'entities/mentions.fields.element',
-                'render' => Standard::ENTITYLINK,
+                'render' => function ($model) {
+                    $row = '<i class="fa-solid fa-lock" title="' . __('crud.is_private') . '" data-toggle="tooltip"></i> ';
+
+                    if ($model->entity && !$model->entity->is_private || 
+                        $model->questElement && $model->questElement->quest && !$model->questElement->quest->entity->is_private || 
```

Comment:
Missing permission check on ->entity, also you need to add `()` around the logical checks or it won't do what you expect ;-)